eth/consensus/clique/clique.py

```diff
@@ -0,0 +1,189 @@
+import logging
+import time
+from typing import Sequence, Iterable
+
+from eth.db.chain import ChainDB
+from eth_typing import (
+    Address,
+    Hash32,
+)
+from eth_utils import (
+    decode_hex,
+    encode_hex,
+    to_tuple,
+    ValidationError,
+)
+
+from eth.rlp.headers import BlockHeader
+
+from eth.typing import VMConfiguration
+from eth.vm.chain_context import ChainContext
+from eth.vm.execution_context import (
+    ExecutionContext,
+)
+# from eth.vm.forks.byzantium.headers import (
+#     create_header_from_parent,
+#     configure_header,
+# )
+
+from .datatypes import (
+    Snapshot,
+)
+from .header_cache import HeaderCache
+from .snapshot_manager import SnapshotManager
+from ._utils import (
+    get_signer_address,
+    humanize_hashes,
+    validate_header_integrity,
+)
+
+# https://github.com/ethereum/EIPs/issues/225
+# https://github.com/ethereum/go-ethereum/blob/0ac9bbba6cd259bc9895e786121cdcb6e7c5d9d2/consensus/clique/clique.go
+
+
+
+def create_execution_context(self,
+                             prev_hashes: Iterable[Hash32],
+                             chain_context: ChainContext,
+                             import_block: BlockHeader=None) -> ExecutionContext:
+
+    # In Clique consensus, the tx fee goes to the signers so for the scope of the
+    # Execution context we set the coinbase to the signer if possible.
+
+    if import_block is not None:
+        try:
+            coinbase = get_signer_address(import_block.header)
+        except Exception:
+            coinbase = self.coinbase
+    else:
+        coinbase = self.coinbase
+
+    return ExecutionContext(
+        coinbase=coinbase,
```

Comment:
rather than overload the `coinbase` I think it would be better to add a new field `ExecutionContext.fee_recipient`.  This value would be optional, defaulting to the `coinbase` unless explicitely provided.  I think you'll run into problems since what you're doing will change the value returned by the `COINBASE` opcode during EVM execution which may or may not be the desired behavior.

Reply:
>I think you'll run into problems since what you're doing will change the value returned by the COINBASE opcode during EVM execution which may or may not be the desired behavior.

That's a good point :+1: